.github/workflows/on_push.yaml

```diff
@@ -0,0 +1,29 @@
+name: Test on push and publish
+
+# On push to a "special" branch, we:
+# * always publish to charmhub at latest/edge/branchname
+# * always run tests
+# where a "special" branch is one of main/master or track/**, as 
+# by convention these branches are the source for a corresponding
+# charmhub edge channel.  
+
+on:
+  push:
+    branches:
+      - master
+      - main
+      - track/**
+
+jobs:
+
+  tests:
+    name: Run Tests
+    runs-on: ubuntu-latest
```

Comment:
I think we can remove this line as it's already specified in `tests.yaml`

Reply:
oh yeah I've got no idea how this works haha.  Feel free to modify all this however you'd like, it is just a suggestion :)